resources/splunk_app.rb

```diff
@@ -32,8 +33,18 @@ def install_state
     node.run_state['splunk_ingredient']['installations'][install_dir]
   end
 
+  def absolute_app_root
+    root_path = case app_root
+                when :master_apps then 'master-apps/apps'
+                when :shcluster then 'shcluster/apps'
+                else app_root
+                end
+
+    Pathname.new(install_dir) + 'etc' + root_path
```

Comment:
It might be worth a comment, that + is join, and that + absolute path means resolve the absolute path because this threw me for a brief loop.